### ISSUE TITLE
Axe accessibility issue with Button inside Antd 4.x Collapse

### ISSUE THREAD

**Issue by abhishekahuja87** [ant-design/ant-design#31123]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/embed/elastic-colden-odgbw?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce
I see accessibility issue when using [Axe](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd) accessibility chrome plugin on my App which has [Collapse](https://ant.design/components/collapse/) from Antd 3.x

[codesandbox link](https://codesandbox.io/embed/elastic-colden-odgbw?fontsize=14&hidenavigation=1&theme=dark)

Issue: Nested interactive controls are not announced by screen readers

Reason: Since there is a button in the accordion

If I remove the button, the above accessibility is solved. But I would like to have the button

I also tried to replace button by span like 

    <span tabIndex="0"><Icon/></span> 

but even this way the accessibility issue persists. But if I remove the 'tabIndex="0"', it works fine.

What is the way to put items in the Accordion which are also accessible via keyboard(hence tabIndex=0).

TIA

### What is expected?
No Accessibility issue

### What is actually happening?
I see the accessibility issue

| Environment | Info |
|---|---|
| antd | 4.16.5 |
| React | Latest |
| System | Any |
| Browser | Any |

**Comment by abhishekahuja87** [comment]
Please reopen. This is also the issue wth 4.x latest version

**Comment by yoyo837** [comment]
Your Reproduction link is 3.x

**Comment by yoyo837** [comment]
Feel free to open a new issue with 4.x demo.

**Comment by afc163** [comment]
@abhishekahuja87 You can try `collapse="header"` to split button and collapse interactive area.
